File: backend/models/Exercise.js
const mongoose = require('mongoose');

const ExerciseSchema = new mongoose.Schema({
    title: { type: String, required: true },
    subtitle: { type: String, required: true },
    goals: [String], // ["Weight Loss", "Muscle Gain", etc]
    equipmentNeeded: [String], // ["Dumbbells", "None", etc]
    location: { type: String, enum: ["Home", "Gym", "Outdoor", "Any"], default: "Any" },
    intensity: { type: String, enum: ["Low", "Medium", "High"] },
    isInjurySafe: { type: Boolean, default: true },
    difficulty: { type: String, enum: ["Beginner", "Intermediate", "Advanced"] },
    steps: [String], // Step-by-step instructions
    description: String,
});

module.exports = mongoose.model('Exercise', ExerciseSchema);

File: frontend/lib/data/models/exercise.dart
class Exercise {
  final String title;
  final String subtitle;
  final List<String> goals;
  final List<String> steps;
  final String category;

  Exercise({
    required this.title,
    required this.subtitle,
    required this.goals,
    required this.steps,
    required this.category,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      title: json['title'] ?? '',
      subtitle: json['subtitle'] ?? '',
      goals: List<String>.from(json['goals'] ?? []),
      steps: List<String>.from(json['steps'] ?? []),
      category: json['category'] ?? '',
    );
  }
}
